template/src/lib/sdk/widgets/GeocoderWidget.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { geocodeStreetPoint } from 'lib/sdk';
+
+import { selectOAuthCredentials } from 'config/oauthSlice';
+import { addLayer, setError, setGeocoderResult, setViewState } from 'config/cartoSlice';
+
+import { CircularProgress, InputBase, Paper } from '@material-ui/core';
+import SearchIcon from '@material-ui/icons/Search';
+import { makeStyles } from '@material-ui/core/styles';
+
+const DEFAULT_COUNTRY = ''; // 'SPAIN', 'USA'
+
+const useStyles = makeStyles((theme) => ({
+  input: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+}));
+
+export default function GeocoderWidget(props) {
+  const oauthCredentials = useSelector(selectOAuthCredentials);
+  const credentials = useSelector((state) => state.carto.credentials);
+  const getDataServicesCredentials = () => {
+    if (oauthCredentials) {
+      return oauthCredentials;
+    }
+    return credentials;
+  };
+
+  // Component local state and events handling
+  const [searchText, setSearchText] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const handleChange = (e) => {
+    setSearchText(e.target.value);
+  };
+
+  const handleInput = (e) => {
+    if (e.target.value === '') {
+      updateMarker(null);
+    }
+  };
+
+  const handleKeyPress = async (e) => {
+    const credentials = getDataServicesCredentials();
```

Comment:
We need a debounce here to save credits

Reply:
The geocoding is just launched on 'Enter' click